### PR TITLE
Feat: Configure elastic training in pytorch plugin

### PR DESCRIPTION
**Pull request by fg91**
# TL;DR
This PR modifies the pytorch plugin so that it can set an [`ElasticPolicy` in the kubeflow `PytorchJob` ](https://github.com/kubeflow/training-operator/blob/8a066f9386e9fb739815e9001f4fe53158227646/pkg/apis/kubeflow.org/v1/pytorch_types.go#L90) in case a user configures [torch elastic training (torchrun)](https://pytorch.org/docs/stable/elastic/run.html) in the task decorator:

```python
from flytekitplugins.kfpytorch import Elastic

@task(
    task_config=Elastic(
        replicas=4,
        nproc_per_node=4,
        ...
    ),
    ...
)
def train(...):
    ...
```

See [this](https://github.com/flyteorg/flyte/issues/3614) issue for motivation and more details.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [x] Any pending items have an associated Issue

## Tracking Issue
Fixes https://github.com/flyteorg/flyte/issues/3614

### CODE REVIEW

**Comment by fg91** [comment]
Tests are failing since flyteidl needs to be updated first.

**Review by kumare3 — APPROVED** [review]
